src/components/Editor/CallSites.js

```diff
@@ -44,18 +52,69 @@ class CallSites extends Component {
     selectedLocation: Object
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = this.getEditorViewport();
+  }
+
   componentDidMount() {
     const { editor } = this.props;
     const codeMirrorWrapper = editor.codeMirror.getWrapperElement();
 
     codeMirrorWrapper.addEventListener("click", e => this.onTokenClick(e));
+    editor.codeMirror.on("scroll", this.onEditorScroll);
   }
 
   componentWillUnmount() {
     const { editor } = this.props;
     const codeMirrorWrapper = editor.codeMirror.getWrapperElement();
 
     codeMirrorWrapper.removeEventListener("click", e => this.onTokenClick(e));
+    editor.codeMirror.off("scroll", this.onEditorScroll);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.selectedSource != this.props.selectedSource) {
+      this.setState(this.getEditorViewport());
+    }
+  }
+
+  onEditorScroll = debounce(e => {
+    this.setState(this.getEditorViewport());
+  }, 200);
+
+  getEditorViewport() {
+    const { editor } = this.props;
+
+    // Get scroll position
+    const charWidth = editor.codeMirror.defaultCharWidth();
+    const scrollArea = editor.codeMirror.getScrollInfo();
+    const { scrollLeft } = editor.codeMirror.doc;
+    const rect = editor.codeMirror.getWrapperElement().getBoundingClientRect();
+    const topVisibleLine = editor.codeMirror.lineAtHeight(rect.top, "window");
+    const bottomVisibleLine = editor.codeMirror.lineAtHeight(
+      rect.bottom,
+      "window"
+    );
+
+    // Update state which should trigger appropriate re-renders
+    const leftColumn = Math.floor(scrollLeft > 0 ? scrollLeft / charWidth : 0);
```

Comment:
Why is there a > 0 check here and for rightCharacter? It seems unnecessary to me since without the check 0 is still 0 and negative values will floor as expected for a Cartesian plane.